libs/designer/src/lib/ui/panel/templatePanel/createWorkflowPanel/usePanelTabs.tsx

```diff
@@ -8,33 +8,52 @@ import { useDispatch, useSelector } from 'react-redux';
 import type { AppDispatch, RootState } from '../../../../core/state/templates/store';
 import type { TemplatePanelTab } from '@microsoft/designer-ui';
 import Constants from '../../../../common/constants';
-import { TemplateService, isUndefinedOrEmptyString } from '@microsoft/logic-apps-shared';
+import { TemplateService } from '@microsoft/logic-apps-shared';
+import { useExistingWorkflowNames } from '../../../../core/queries/template';
+import {
+  validateConnections,
+  validateKind,
+  validateParameters,
+  validateWorkflowName,
+} from '../../../../core/state/templates/templateSlice';
 
 export const useCreateWorkflowPanelTabs = ({ onCreateClick }: { onCreateClick: () => Promise<void> }): TemplatePanelTab[] => {
   const intl = useIntl();
   const dispatch = useDispatch<AppDispatch>();
-  const { workflowName, kind, parameters, manifest: selectedManifest } = useSelector((state: RootState) => state.template);
-  const { existingWorkflowName } = useSelector((state: RootState) => state.workflow);
+  const { data: existingWorkflowNames } = useExistingWorkflowNames();
+  const {
+    errors: { workflow: workflowError, kind: kindError, parameters: parameterErrors, connections: connectionsError },
+    kind,
+    manifest: selectedManifest,
+  } = useSelector((state: RootState) => state.template);
+  const { mapping } = useSelector((state: RootState) => state.workflow.connections);
+  const selectedTabId = useSelector((state: RootState) => state.panel.selectedTabId);
   const [isLoadingCreate, setIsLoadingCreate] = useState(false);
   const [isCreated, setIsCreated] = useState(false);
 
   const connectionsExist = useMemo(() => selectedManifest && Object.keys(selectedManifest?.connections).length > 0, [selectedManifest]);
   const parametersExist = useMemo(() => selectedManifest && selectedManifest.parameters.length > 0, [selectedManifest]);
-  const hasParametersValidationErrors = useMemo(
-    () => Object.values(parameters.validationErrors).some((error) => !!error),
-    [parameters.validationErrors]
-  );
-  const missingRequiredParameters = useMemo(
-    () => Object.values(parameters.definitions).some((param) => param.required && !param.value),
-    [parameters.definitions]
-  );
-  const missingWorkflowName = isUndefinedOrEmptyString(existingWorkflowName ?? workflowName);
+  const hasParametersValidationErrors = useMemo(() => Object.values(parameterErrors).some((error) => !!error), [parameterErrors]);
 
   useEffect(() => {
     setIsLoadingCreate(false);
     setIsCreated(false);
   }, [selectedManifest]);
 
+  useEffect(() => {
+    dispatch(validateConnections(mapping));
```

Comment:
Does this mean we will see error icon on teh connections tab when we first open it from create workflow button?

Reply:
Discussed offline;
The error icon does not show on the connections tab when we first open since the connection tab will always be the first (selected doesn't show error icon).

To keep the logic extra clean and organized, added an extra if condition to run validationConnection only on non-connection tabs